layers/+distributions/spacemacs-base/packages.el

```diff
@@ -240,9 +240,20 @@
     :init (spacemacs/set-leader-keys "ji" 'imenu)))
 
 (defun spacemacs-base/init-linum ()
-  (when dotspacemacs-line-numbers
-    (add-hook 'prog-mode-hook 'linum-mode)
-    (add-hook 'text-mode-hook 'linum-mode))
+  (use-package linum
+    :config
+    (progn
+      (if (or (eq dotspacemacs-line-numbers t)
+              (eq dotspacemacs-line-numbers 'relative))
+          (progn
+            (add-hook 'prog-mode-hook 'linum-mode)
+            (add-hook 'text-mode-hook 'linum-mode)))
+      (defun linum-on ()
+        "Overwrite the original `linum-on' function with a more selective one."
+        (when (spacemacs/linum-enabled-p)
+          (linum-mode 1)))
```

Comment:
I guess, didn't know for sure so I called it exactly as it was called in the original `linum-on` function in `linum.el:116`